community/kernel/src/main/java/org/neo4j/kernel/api/txstate/TransactionState.java

```diff
@@ -85,12 +86,11 @@
 
     void constraintDoAdd( ConstraintDescriptor constraint );
 
-    void constraintDoAdd( IndexBackedConstraintDescriptor constraint, long indexId );
+    void constraintDoAdd( IndexBackedConstraintDescriptor constraint, IndexRule indexRule );
```

Comment:
Thinking about the near future, so apply pinch of salt logic: Having IndexRule here ties TransactionState, which is logical in nature, a bit more to the physical realm. This may be that there's a missing abstraction between indexId and indexRule somewhere, but I came to think about it now.